test/extensions/common/aws/credentials_provider_impl_test.cc

```diff
@@ -2311,6 +2311,38 @@ TEST_F(WebIdentityCredentialsProviderTest, FullCachedCredentialsWithMissingExpir
 }
 
 TEST_F(WebIdentityCredentialsProviderTest, RefreshOnNormalCredentialExpiration) {
+  // Setup timer.
+  timer_ = new NiceMock<Event::MockTimer>(&context_.dispatcher_);
+  // Time 2018-01-02T05:04:05Z in unix_timestamp is 1.514869445E9
+  expectDocument(200, std::move(R"EOF(
+{
+  "AssumeRoleWithWebIdentityResponse": {
+    "AssumeRoleWithWebIdentityResult": {
+      "Credentials": {
+        "AccessKeyId": "akid",
+        "SecretAccessKey": "secret",
+        "SessionToken": "token",
+        "Expiration": 1.514869445E9
+      }
+    }
+  }
+}
+)EOF"));
+  setupProviderWithContext();
+  timer_->enableTimer(std::chrono::milliseconds(1), nullptr);
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(std::chrono::hours(2)), nullptr));
+
+  // Kick off a refresh
+  timer_->invokeCallback();
+
+  const auto credentials = provider_->getCredentials();
+  EXPECT_EQ("akid", credentials.accessKeyId().value());
+  EXPECT_EQ("secret", credentials.secretAccessKey().value());
+  EXPECT_EQ("token", credentials.sessionToken().value());
+}
+
+TEST_F(WebIdentityCredentialsProviderTest, RefreshOnNormalCredentialExpirationIntegerFormat) {
   // Setup timer.
   timer_ = new NiceMock<Event::MockTimer>(&context_.dispatcher_);
   // Time 2018-01-02T05:04:05Z in unix_timestamp is 1514869445
```

Comment:
Is it possible to get integer expiration format line in L2357 ?
Are we making the functionality to handle both?

Reply:
If we are sure STS json response will always contain "scientific notation" then we can get rid of `RefreshOnNormalCredentialExpirationIntegerFormat` test case, assume that as bad format and ignore the expiration value and depend on default refresh period of 1 hour.

Reply:
utility.cc will handle both cases. I don't expect that STS will put a fix in for normal integers any time soon, but if they do we will continue to function correctly.

i put the RefreshOnNormalCredentialExpirationIntegerFormat test case to ensure both scenarios will always work as expected